BX-4288 Add support for multiple endpoints

DimaCantemir opened the pull request:
Added support for GET /Orders
Added support for GET /Trades
Added support for GET /Tickers

Added support for Trades feed

DimaCantemir committed:
BX-4288
Added support for GET /Orders
Added support for GET /Trades
Added support for GET /Tickers

Added support for Trades feed

kdonthi commented on bxserum/provider_test/integration/http_test.go:
```diff
@@ -36,4 +37,23 @@ func TestHTTPClient_Requests(t *testing.T) {
+
+	testGetOrders(
+		t,
+		func(ctx context.Context, market string, owner string) *pb.GetOrdersResponse {
```
Can you add a function to handle error cases for the functions?

kdonthi commented on examples/httpclient/main.go:
```diff
@@ -4,13 +4,17 @@ import (
 
 func main() {
-	callHTTP()
+	callOrderbookHTTP()
```
instead of `call`, I think it might be nicer to add the method type like `getOrderbookHTTP`?

kdonthi commented on bxserum/provider_test/integration/public.go:
```diff
@@ -116,3 +132,27 @@ func assertMarketNotPresent(t *testing.T, marketName string, markets *pb.GetMark
+
+func assertOrder(t *testing.T, expectedName string, orders *pb.GetOrdersResponse) {
+	require.NotEmpty(t, orders.GetOrders())
+	order := *(orders.Orders[0])
```
If you are dereferencing it, would it also be a good idea to check that orders.Orders[0] is not nil?

kdonthi reviewed (approved):
Looks good for the most part, and it would be nice to add some error cases for the tests. I like the separation of the different methods in the sample code

aspin commented on bxserum/provider/http.go:
```diff
@@ -28,7 +33,7 @@ func NewHTTPTestnet() *HTTPClient {
-func NewHTTPClientWithEndpoint(endpoint string, client *http.Client) *HTTPClient {
+func NewHTTPClientWithEndpoint(endpoint string, client *http.Client, timeout time.Duration) *HTTPClient {
 	if client == nil {
 		client = &http.Client{Timeout: time.Second * 7}
```
use time out here?

DimaCantemir committed:
BX-4288
Added support for GET /Orders
Added support for GET /Trades
Added support for GET /Tickers

updates after review
